scripts/setup.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+set -e
+
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+RED='\033[0;31m'
+BLUE='\033[0;34m'
+NC='\033[0m'
+
+print_status() {
+  local type=$1
+  local message=$2
+  local icon
+  local color
+
+  case $type in
+        "task")     icon="📦"; color=$YELLOW ;;
+        "error")    icon="❌"; color=$RED ;;
+        "success")  icon="✅"; color=$GREEN ;;
+        "warning")  icon="⚠️ "; color=$YELLOW ;;
+        "info")     icon="ℹ️ "; color=$BLUE ;;
+        *)          icon="📦"; color=$YELLOW ;;
+    esac
+
+  echo -e "\n$icon ${color}$message${NC} ..."
+}
+
+command_exists() {
+  command -v "$1" >/dev/null 2>&1
+}
+
+# Check/Install NVM
+if [ ! -d "$HOME/.nvm" ]; then
+  print_status "task" "Installing NVM"
+  PROFILE=/dev/null curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash
+
```

Comment:
I'm not really sure that I want this script to auto-install NVM, I think it should rather let the user install it himself and just exit the script with a message to install it himself if he doesn't already have it. 
I understand that the curl URL looks fine but from my perspective, it's a security risk and would rather not do that